lib/IRGen/GenDistributed.cpp

```diff
@@ -271,17 +271,20 @@ static CanSILFunctionType getAccessorType(IRGenModule &IGM,
     auto *actor = getDistributedActorOf(Target);
     assert(actor);
 
-    for (auto *genericParam : actor->getInnermostGenericParamTypes())
-      genericParams.push_back(genericParam);
-
+//    for (auto *genericParam : actor->getInnermostGenericParamTypes())
+//      genericParams.push_back(genericParam);
+//
     // Add a generic parameter `D` which stands for decoder type in the
     // accessor signature - `inout D`.
     genericParams.push_back(decoderType);
     // Add a requirement that decoder conforms to the expected protocol.
     genericRequirements.push_back(
         {RequirementKind::Conformance, decoderType, decoderProtocolTy});
 
-    signature = GenericSignature::get(genericParams, genericRequirements);
+//    signature = GenericSignature::get(genericParams, genericRequirements);
+    signature = buildGenericSignature(Context, actor->getGenericSignature(),
```

Comment:
Can you try changing actor->getGenericSignature() here into GenericSignature()?

Reply:
done, and yeah we finally figured it out :)